File: materials/tdd/Java/simple_example/simple_example.java
import java.util.*;

class ListProperties {

  public int GetMaximumValueFromList (List<Integer> List) {

    int Result = List.get(0);

    for (int iterator = 0; iterator < List.size(); iterator++)
      if (Result < List.get(iterator))
        Result = List.get(iterator);

    return Result;
  }

  public boolean CheckExistingValueInList (List<Integer> List, int ValueToCheck) {

    for (int iterator = 0; iterator < List.size(); iterator++)
      if (ValueToCheck == List.get(iterator))
        return true;

    return false;
  }
}

class ListBuilder {

  public void AssignList (List<Integer> List) {

    List.add(1);
    List.add(2);
    List.add(3);
    List.add(4);
    List.add(5);
    List.add(6);
    List.add(7);

  }
}

class TestCases {

  public void TestGetMaximumValueFromList (List<Integer> List, ListProperties Properties) {

    if (Properties.GetMaximumValueFromList(List) == 7)
      System.out.println("Test passed");

    else
      System.out.println("Test failed");
  }

  public void TestCheckExistingValueInList (List<Integer> List, ListProperties Properties) {

    if (Properties.CheckExistingValueInList(List, 4) == true)
      System.out.println("Test passed");

    else
      System.out.println("Test failed");
  }
}

class MainWorkFlow {

  public static void main (String[] args) {

    ListProperties Properties = new ListProperties();
    ListBuilder Builder = new ListBuilder();
    TestCases Cases = new TestCases();

    List<Integer> List = new ArrayList<Integer>();

    Builder.AssignList(List);

    Cases.TestGetMaximumValueFromList(List, Properties);
    Cases.TestCheckExistingValueInList(List, Properties);
  }
}

File: materials/tdd/Python/simple_example/simple_example.py
class lists_properties():
    """properties for lists."""

    def __init__(self):
        super(lists_properties, self).__init__()


    def get_maximum_value_from_list(self, list):

        maximum_value = list[0];

        for item in list:
            if maximum_value < item:
                maximum_value = item

        return maximum_value


    def check_existing_value_in_list(self, list, value_to_check):

        for item in list:
            if item == value_to_check:
                return True

        return False

class test_cases():
    """test cases for lists_properties methods."""

    def __init__(self):
        super(test_cases, self).__init__()


    def assert_get_maximum_value_from_list(self, properties):

        the_list = [12, -12, 13, 72, 92, -1002, -345, 5421]

        assert properties.get_maximum_value_from_list(properties, the_list) == 5421


    def assert_check_existing_Value_in_list(self, properties):

        the_list = [12, -12, 13, 72, 92, -1002, -345, 5421]

        assert properties.check_existing_value_in_list(properties, the_list, -1002) == True

properties = lists_properties
cases = test_cases

cases.assert_get_maximum_value_from_list(cases, properties)
cases.assert_check_existing_Value_in_list(cases, properties)
